fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/values/RelOpValue.java

```diff
@@ -184,6 +184,13 @@ private static Optional<QueryPredicate> promoteOperandsAndCreatePredicate(@Nulla
                                                                               @Nonnull Value leftChild,
                                                                               @Nonnull Value rightChild,
                                                                               @Nonnull final Comparisons.Type comparisonType) {
+        if (leftChild.getResultType().getTypeCode() == Type.TypeCode.NULL && rightChild.getResultType().getTypeCode() == Type.TypeCode.NULL) {
```

Comment:
I'm a little confused of the difference between `TypeCode.Unknown` and `TypeCode.NULL`, should we use `NULL` or `Unknown` here?

Reply:
so the reason that the code is here, is because of the check at line #196: `final var maxtype = Verify.verifyNotNull(Type.maximumType(leftChild.getResultType(), rightChild.getResultType()));`. It'll raise an error for queries like `where null is distinct from null`. Do we want to remove the check at line#196 instead?

Reply:
The maximum type of two NullTypes should be the NullType. So maximumType seems to not do the right thing.